Add microsoft-azure-storage-connector-example (#1400)
wso2/docs-ei: en/micro-integrator/docs/references/connectors/microsoft-azure-storage-connector/microsoft-azure-storage-connector-example.md

# Microsoft Azure Storage Connector

The Microsoft Azure Storage Connector allows you to access the Azure Storage services using Microsoft Azure Storage Java SDK through WSO2 EI. [Azure Storage](https://azure.microsoft.com/en-us/) is a Microsoft-managed cloud service that provides storage that is highly available, secure, durable, scalable and redundant. The Azure Storage consists of four primary Azure Storage types. They are blob storage, table storage, file storage, queue storage.

> Given below is a sample scenario that demonstrates how working with containers and blobs operations using the WSO2 Microsoft Azure Storage Connector.

Reviewer: ```suggestion
Given below is a sample scenario that demonstrates how to work with container and blob operations using the WSO2 Microsoft Azure Storage Connector.
```

Author: Updated.

## What you'll build

This example demonstrates how to use Microsoft Azure Storage connector to:

1. Create a container (a location for storing employee details) in Microsoft Azure Storage account.
2. Retrieve information about the created containers.
3. Upload text or binary employee details (blob) in to the container.
4. Retrieve information about the uploaded employee details(blob).
5. Remove uploaded employee details (blob).
6. Remove created container.

All six operations are exposed via an API. The API with the context `/resources` has six resources  

* `/createcontainer` : Creates a new container in the Microsoft Azure Storage account with the specified container name for store employee details.
* `/listcontainer` : Retrieve information about the created containers from the Microsoft Azure Storage account.
* `/adddetails`: Upload text or binary employee data (blob) and stored into the specified container.
* `/listdetails` : Retrieve information about the added employee deta (blobs).
* `/deletedetails` : Remove added employee data from the specified text or binary employee data (blob).
* `/deletecontainer` : Remove created container in the Microsoft Azure Storage account.

> To know the further information about these operations please refer the [Microsoft Azure Storage connector reference guide](microsoft-azure-storage-connector-reference.md).

Reviewer: ```suggestion
For more information about these operations, please refer to the [Microsoft Azure Storage connector reference guide](microsoft-azure-storage-connector-reference.md).
```

Author: Updated.

> > **Note**: Before invoke the API you need to create a **Storage Account** in **Microsoft Azure Storage account**, see [Azure Storage Configuration](microsoft-azure-storage-configuration.md) documentation.

Reviewer: ```suggestion
> **Note**: Before invoking the API, you need to create a **Storage Account** in **Microsoft Azure Storage account**. See [Azure Storage Configuration](microsoft-azure-storage-configuration.md) documentation for more information.
```

Author: Updated.

> Following diagram shows the overall solution. User creates a container, stores some text or binary employee data (blob) into the container and then he receives it back. To invoke each operation user uses the same API. 

Reviewer: ```suggestion
The following diagram shows the overall solution. The user creates a container, stores some text or binary employee data (blob) into the container, and then receives it back. To invoke each operation, the user uses the same API. 
```

Author: Updated.

       </resource>
       <resource methods="POST" url-mapping="/deletecontainer">
           <inSequence>
               <property expression="json-eval($.accountName)" name="accountName" scope="default" type="STRING"/>
               <property expression="json-eval($.accountKey)" name="accountKey" scope="default" type="STRING"/>
               <property expression="json-eval($.containerName)" name="containerName" scope="default" type="STRING"/>
               <msazurestorage.init>
                   <accountName>eiconnectortest</accountName>
                   <accountKey>bWt69gFpheoD6lwVsMgeV5io2/KxlXK1KUcod68PhzuV1xHxje0LBD4Bd+y+ESAOlH5BTAfvdDG5q4Hhg==</accountKey>
               </msazurestorage.init>
               <msazurestorage.deleteContainer>
                   <containerName>{$ctx:containerName}</containerName>
               </msazurestorage.deleteContainer>
               <log level="full">
                   <property name="Delete selected container" value="Delete selected container"/>
               </log>
               <respond/>
           </inSequence>
           <outSequence/>
           <faultSequence/>
       </resource>
   </api>
   
   ```

> **Note :** Please modify below properties of the configuration as applicable.

Reviewer: ```suggestion
**Note**: Please modify the following properties of the configuration as applicable.
```

Author: Updated.
